src/builtins.c

```diff
@@ -1409,7 +1411,7 @@ size_t jl_static_show_x(JL_STREAM *out, jl_value_t *v, int depth)
         n += jl_printf(out, ":%s", ((jl_sym_t*)v)->name);
     }
     else if (jl_is_gensym(v)) {
-        n += jl_printf(out, "GenSym(%d)", ((jl_gensym_t*)v)->id);
+        n += jl_printf(out, "GenSym(%d)", (int)((jl_gensym_t*)v)->id);
```

Comment:
What is the type of id here?  Could that ever be larger than 2^31? Should this maybe be something like:
`"GenSym(" FORMAT_FOR_ID ")", ((jl_gensym_t*)v)->id`